libnopegl/src/pipeline.c

```diff
@@ -31,6 +71,9 @@ struct pipeline *ngli_pipeline_create(struct gpu_ctx *gpu_ctx)
 
 int ngli_pipeline_init(struct pipeline *s, const struct pipeline_params *params)
 {
+    int ret = pipeline_layout_copy(&s->layout, &params->layout);
+    if (ret < 0)
+        return ret;
     return s->gpu_ctx->cls->pipeline_init(s, params);
```

Comment:
So we're making a copy of the layout for a future use within this pipeline module, but we're still using the original layout in gl/vk pipelines mmh. Wouldn't it make sense to dup the whole params structure within the pipeline module and pass this copy down to the backend specific implementations?

Reply:
I only need the layout for now  (and for the upcoming PRs), this is why i haven't copied the whole params.

Reply:
I moved the layout field below the program one in the pipeline structure so it matches the order in the pipeline_params structure. I also moved the copy of the layout directly inside pipeline_gl/vk where the other fields are already copied so it is more consistent with the current code while not excluding replacing those fields with a pipeline_params structure a future pr. I prefered to do it this way for now and not touch the pipeline_gl/vk_init prototypes (so they still match their parent prototype).